### DISCUSSION TITLE
API works when using fetch hook. Same API doesn't work using @click method

### DISCUSSION THREAD

**Discussion by mhabbasi-gh** [nuxt/nuxt#8992] [Questions (Nuxt 2)]
In my Nuxt app, using Axios I fetch the same API once in `fetch()` hook and then once in a method in my Nuxt page. When using `fetch()` the API works and I get the correct response, however when I fetch it through a method after the page is rendered I get `Cross-origin resource sharing error:PreFlightMissingAllowOriginHeader` error! What am I doing wrong here?

Also this is the error I get in console when I fetch it using methods:
```
...has been blocked by CORS policy: Response to preflight request doesn't
pass access control check: No 'Access-Control-Allow-Origin' header is
present on the requested resource.
```

**Comment by jasonvarga** [comment]
I just ran into something similar and arrived at this discussion. It sounds like my issue.

I don't know what backend you were using, but for me it was a Laravel application.
I didn't have the `HandleCors` middleware in my Kernel, so the preflight requests were failing as they were missing the header.
